src/lib.rs

```diff
@@ -1,71 +1,75 @@
-use std::mem;
-use std::prelude::v1::*;
+// This supplements the `futures-preview` provided `chunks` operator[1] with a
+// an additional delay to flush the inner vector.
+//
+// [1]: https://github.com/rust-lang-nursery/futures-rs/blob/4613193023dd4071bbd32b666e3b85efede3a725/futures-util/src/stream/chunks.rs
+
+use core::mem;
+use core::pin::Pin;
+use futures::stream::{Fuse, FusedStream, Stream};
+use futures::task::{Context, Poll};
+use futures::StreamExt;
+#[cfg(feature = "sink")]
+use futures_sink::Sink;
+use pin_utils::{unsafe_pinned, unsafe_unpinned};
+
+use futures01::Async;
 use std::time::{Duration, Instant};
-
-use futures::stream::{Fuse, Stream};
-use futures::{Async, Future, Poll};
-use tokio::timer;
+use tokio::prelude::Future;
 use tokio::timer::Delay;
+//
+impl<T: ?Sized> ChunksTimeoutStreamExt for T where T: Stream {}
+
+pub trait ChunksTimeoutStreamExt: Stream {
+    fn chunks_timeout(self, capacity: usize, duration: Duration) -> ChunksTimeout<Self>
+    where
+        Self: Sized,
+    {
+        ChunksTimeout::new(self, capacity, duration)
+    }
+}
+//
 
-/// An adaptor that chunks up elements in a vector.
-///
-/// This adaptor will buffer up a list of items in the stream and pass on the
-/// vector used for buffering when a specified capacity has been reached
-/// or a predefined timeout was triggered.
-///
-/// This was taken and adjusted from
-/// https://github.com/alexcrichton/futures-rs/blob/master/src/stream/chunks.rs
-/// and moved into a separate crate for usability.
+#[derive(Debug)]
 #[must_use = "streams do nothing unless polled"]
-pub struct Chunks<S>
-where
-    S: Stream,
-{
+pub struct ChunksTimeout<St: Stream> {
```

Comment:
Did you rename that to `ChunksTimeout` because `Chunks` is already taken by `futures`?

Reply:
Yes, I thought that was clearer. But `Chunks` is also fine, as there is now the added combinator `chunks_timeout`.
I just thought it would be more ergonomic to have a distinct name; I'm open to suggestions on the name though.

Reply:
I like that it's more explicit about what it's doing, so yeah I like it.
Although it sounds a bit like it only defines the timeout **for** the chunks whereas in reality it returns a chunk **after** a timeout. Was considering `ChunksWithTimeout`, but it doesn't have the same ring to it. Let's go with `ChunksTimeout` then.